delta-app/src/main/java/io/cdap/delta/app/service/AssessmentHandler.java

```diff
@@ -161,10 +192,18 @@ public void assessDraft(HttpServiceRequest request, HttpServiceResponder respond
                           @PathParam("draft") String draftName) {
     respond(namespaceName, responder, ((namespace) -> {
       DraftId draftId = new DraftId(namespace, draftName);
-      PluginConfigurer pluginConfigurer = getContext().createPluginConfigurer(namespaceName);
-      PipelineAssessment assessment =
-        getDraftService().assessPipeline(draftId, new DefaultConfigurer(pluginConfigurer));
-      responder.sendString(GSON.toJson(assessment));
+      Draft draft = getDraftService().getDraft(draftId);
+
+      draft.getConfig().validatePipeline();
```

Comment:
Moved here since its better to fail validation early before sending task to worker

Reply:
Does the draft validation perform any validation which may run plugin-specific code? I traced some of the code and noticed this calls into the `Plugin#validate()` method, but I'm unclear on whether that could be overridden or not given it's in the proto module.

Reply:
I think it should not run any plugin specific code, as `Plugin` is just a metadata class to represent it's properties in the draft or app spec https://github.com/data-integrations/delta/blob/develop/delta-proto/src/main/java/io/cdap/delta/proto/Plugin.java

Reply:
Sounds good, thanks!